app/code/local/Magmodules/Webwinkelconnect/controllers/IndexController.php

```diff
@@ -48,4 +48,37 @@ public function indexAction()
         }
     }
 
+    /**
+     * @throws Exception
+     */
+    public function syncAction() {
+        $request_data = trim(file_get_contents('php://input'));
+        $helper = Mage::helper('webwinkelconnect');
+        if (!$request_data) {
+            $helper->returnResponseCode(400, 'Empty request data');
+        }
+        if (!$request_data = json_decode($request_data, true)) {
+            $helper->returnResponseCode(400, 'Invalid JSON data provided');
+        }
+
+
+        if (
+            !$helper->hasCredentialFields($request_data['webshop_id'], $request_data['api_key'])
+            || $helper->credentialsEmpty($request_data['webshop_id'], $request_data['api_key'])
+        ) {
+            $helper->returnResponseCode(403,'Missing credential fields');
+        }
+
+        $helper->isAuthorized($request_data['webshop_id'], $request_data['api_key']);
+
+        if (empty(Mage::getResourceSingleton('catalog/product')->getProductsSku([$request_data['product_review']['product_id']]))) {
```

Comment:
Wait. Here you're searching by SKU right? But later you're using the same value as the product primary key (ID). Is that correct? Is this well tested with different products and different IDs/SKUs?